core/storage/skill/gae_models.py

```diff
@@ -150,3 +150,30 @@ class SkillSummaryModel(base_models.BaseModel):
     # model was created).
     skill_model_created_on = ndb.DateTimeProperty(required=True, indexed=True)
     version = ndb.IntegerProperty(required=True)
+
+
+class SkillsMasteryModel(base_models.BaseModel):
+    """Mastery model of a user for Oppia Skill.
+
+    This model stores progress of each skill for a given user
+    and returns skill id.
+    """
+
+    # The user id of the currently logged in user.
+    user_id = ndb.StringProperty(required=True, indexed=True)
+    # The skill id for which degree of mastery is stored.
+    skill_id = ndb.StringProperty(required=True, indexed=True)
+    # The degree of mastery of the user in a skill.
+    degree_of_mastery = ndb.FloatProperty(required=True, indexed=True)
+
+    @classmethod
+    def get_user_mastery(cls, user_id):
+        """Returns skill summary for a particular user.
```

Comment:
? 

This doesn't make sense. What is a "skill summary"?

Probably want to rename to get_all_skill_mastery_models_for_user(cls, user_id) and write an appropriate docstring.

Reply:
No, these are not skill summaries. "Summary" has a technical meaning in Oppia -- see e.g. exp_summary, collection_summary, etc. I would avoid using the term "summary" where it doesn't accurately describe the object/model.

Reply:
Done